config/config.go

```diff
@@ -214,6 +214,7 @@ type IstioConfig struct {
 	ConfigMapName                     string              `yaml:"config_map_name,omitempty"`
 	EnvoyAdminLocalPort               int                 `yaml:"envoy_admin_local_port,omitempty"`
 	GatewayAPIClassName               string              `yaml:"gateway_api_class_name,omitempty"`
+	IstioApiEnabled                   bool                `yaml:"istio_api_enabled,omitempty"`
```

Comment:
This option is just related to the registry right? You still get controlplane information even with this disabled? If so, I think it should go under the `Registry` config and then be a bool `Enabled` to be consistent with the other options like `Tracing.Enabled`.

Reply:
Even if istiod access is disabled, I think we can at least infer some controlplane info from different objects in the kube cluster if istiod is deployed there such as the istiod deployment itself or the sidecar injection webhook.

Reply:
I think it would make sense if we are getting some more controlplane info

Reply:
Given the explanation below, I am now fine with this as-is.  Not sure if @nrfox agrees, perhaps he can comment.

Reply:
I was suggesting `ExternalServices.IstioConfig.Registry.Enabled` since then it'd be consistent with the other external service configs like `ExternalServices.Tracing.Enabled`. Some fields in `IstioConfig` are still relevant even with the registry disabled such as `GatewayAPIClassName` so having a top level `Enabled: false` field would be confusing if other fields in that config are still relevant. If consistency with the other external services is not important since the Istio external service is more unique anyways then I'm good with keeping `IstioApiEnabled`. What do y'all think?

One minor nit if we're keeping `IstioApiEnabled` it should be `IstioAPIEnabled`.